modules/fbs/fbs.module

```diff
@@ -126,15 +126,23 @@ function fbs_service($agency_id = NULL, $endpoint = NULL, $client = NULL, $seria
   // No drupal_static. We want to be callable from tests.
   static $service;
   if (!$service || $reset) {
-    $agency_id = !is_null($agency_id) ? $agency_id : variable_get('fbs_agency', '');
-    $endpoint = !is_null($endpoint) ? $endpoint : variable_get('fbs_endpoint', '');
+    $agency_id = !empty($agency_id) ? $agency_id : variable_get('fbs_agency', '');
+    $endpoint = !empty($endpoint) ? $endpoint : variable_get('fbs_endpoint', '');
+
     if ($endpoint && preg_match('{localtest}', $endpoint)) {
       $client = new FBSFakeHttpClient();
     }
 
     // Ensure exactly one trailing slash.
     $endpoint = rtrim($endpoint, '/') . '/';
-    $service = new FBS($agency_id, $endpoint, $client, $serializer);
+    try {
+      $service = new FBS($agency_id, $endpoint, $client, $serializer);
+    }
+    catch (FBSException $e) {
+      watchdog_exception('fbs', $e);
+
+      return FALSE;
```

Comment:
As all service calls has been wrapped in try blocks, there's no need to catch errors here. Simply move the fbs_service call inside the try block everywhere.